modules/base/rendering/renderabletimevaryingsphere.cpp

```diff
@@ -0,0 +1,554 @@
+        std::optional<bool> useAdditiveBlending;
+
+        // [[codegen::verbatim(MirrorTextureInfo.description)]]
+        std::optional<bool> mirrorTexture;
+
+        // [[codegen::verbatim(FadeOutThresholdInfo.description)]]
+        std::optional<float> fadeOutThreshold [[codegen::inrange(0.0, 1.0)]];
+
+        // [[codegen::verbatim(FadeInThresholdInfo.description)]]
+        std::optional<float> fadeInThreshold;
+
+        // [[codegen::verbatim(DisableFadeInOutInfo.description)]]
+        std::optional<bool> disableFadeInOut;
+
+        // [[codegen::verbatim(BackgroundInfo.description)]]
+        std::optional<bool> background;
+    };
+#include "renderabletimevaryingsphere_codegen.cpp"
+} // namespace
+
+namespace openspace {
+
+double extractTriggerTimeFromFileName(const std::string& filePath);
+
+documentation::Documentation RenderableTimeVaryingSphere::Documentation() {
+    return codegen::doc<Parameters>("base_renderable_time_varying_sphere");
+}
+
+RenderableTimeVaryingSphere::RenderableTimeVaryingSphere(
+                                                      const ghoul::Dictionary& dictionary)
+    : Renderable(dictionary)
+    , _textureSourcePath(TextureSourceInfo)
+    , _orientation(OrientationInfo, properties::OptionProperty::DisplayType::Dropdown)
+    , _size(SizeInfo, 1.f, 0.f, 1e35f)
+    , _segments(SegmentsInfo, 8, 4, 1000)
+    , _mirrorTexture(MirrorTextureInfo, false)
+    , _useAdditiveBlending(UseAdditiveBlendingInfo, false)
+    , _disableFadeInDistance(DisableFadeInOutInfo, true)
+    , _backgroundRendering(BackgroundInfo, false)
+    , _fadeInThreshold(FadeInThresholdInfo, -1.f, 0.f, 1.f)
+    , _fadeOutThreshold(FadeOutThresholdInfo, -1.f, 0.f, 1.f)
+{
+    const Parameters p = codegen::bake<Parameters>(dictionary);
+
+    addProperty(_opacity);
+    registerUpdateRenderBinFromOpacity();
+
+    _size = p.size;
+    _segments = p.segments;
+    _textureSourcePath = p.textureSource;
+
+    _orientation.addOptions({
+        { static_cast<int>(Orientation::Outside), "Outside" },
+        { static_cast<int>(Orientation::Inside), "Inside" },
+        { static_cast<int>(Orientation::Both), "Both" }
+    });
+
+    if (p.orientation.has_value()) {
+        switch (*p.orientation) {
+            case Parameters::Orientation::Inside:
+                _orientation = static_cast<int>(Orientation::Inside);
+                break;
+            case Parameters::Orientation::Outside:
+                _orientation = static_cast<int>(Orientation::Outside);
+                break;
+            case Parameters::Orientation::Both:
+                _orientation = static_cast<int>(Orientation::Both);
+                break;
+            default:
+                throw ghoul::MissingCaseException();
+        }
+    }
+    else {
+        _orientation = static_cast<int>(Orientation::Outside);
+    }
+    addProperty(_orientation);
+
+    _size.setViewOption(properties::Property::ViewOptions::MinMaxRange);
```

Comment:
Size could benefit from being an exponential slider, because of its large max value :)This is done using the `setExponent` function of the property:

` _size.setExponent(15.f);`

The actual suitable exponent depends on the max value, but I something around 15 should be fine here, or maybe a little higher. (The higher the max value, the higher the exponent should be)